### ISSUE TITLE
Timestamp in debug message? 

### ISSUE THREAD

**Issue by moritzraho** [adobe/aio-lib-core-logging#5]
Wdyt?

**Comment by shazron** [comment]
Definitely. I thought it was already in there. 
```
new Date().toISOString()
```

**Comment by Himavanth** [comment]
My original plan was to remove DebugLogger because it really doesn't fit here as a complete logging provider. It doesn't have a bunch of stuff like log levels, filtering based on levels and built-in libraries for basic transports. It's use is best for just debug statements as the name suggests. I have some more digging to do on the custom redirection of its logs and then take the decision of either removing it or enhancing it.

**Comment by moritzraho** [comment]
@Himavanth I think debug logs are useful for libraries, in libraries you don't want to log things unless you are debugging.
Maybe we can get rid of the debuglogger if winston supports a debug mode that works in a similar manner?
Also (other topic) we should do some more investigation on how to enable debug logs in an action as you cannot set env vars from outside of the action. Currently the only solution is  to change the action code (i.e. by adding `process.env['DEBUG'] = 'mymodule'`) which is not practical

**Comment by Himavanth** [comment]
@moritzraho Winston does have debug mode as one of several log levels.

**Comment by moritzraho** [comment]
ok but is there a way for the user of the library to turn debug logs on/off like in the debug module

**Comment by shazron** [comment]
Right now you would do it like this, to only show `<applabel>:info`:
```
DEBUG=<applabel>:info
```
To turn other debug levels on, you add them as comma separated values:
```
DEBUG=<applabel>:info,<applabel>:error
```

However, this is not how debug levels should work. The levels should accumulate. For example, the `silly` level should show the logs for the other levels. Something like: https://www.npmjs.com/package/debug-levels

**Comment by Himavanth** [comment]
@shazron Correct. Debug package is very rudimentary and does not have this feature (and others like custom transports) which means we would have to build it ourselves (it's not a lot of effort though). I spent the first part of this week trying to figure out if it makes sense for us to support it. I have built a custom transport for sending logs to a runtime action here https://github.com/adobe/aio-lib-core-logging/tree/logcollector. Looks like it's ok to give it a try building some of the features in debug package.

**Comment by purplecabbage** [comment]
Is this still an issue?

**Comment by moritzraho** [comment]
Not sure, I would say yes, but then as @Himavanth mentioned we might end up having to add many other features to the debug provider.
Otherwise instead of enriching the debug provider, we can deprecate it and implement tthe `DEBUG=` env var into the winston provider.
I think support for ther `DEBUG=` is a nice to have as it is used by many libraries in the node world.
@shazron @Himavanth  wdyt?

**Comment by meryllblanchet** [comment]
@moritzraho @shazron @Himavanth any conclusion on this one?

**Comment by Himavanth** [comment]
debug npm does not log timestamps when stdout is tty but it does if stdout is say a file. Looks like it is pretty much hardcoded in that aspect. A workaround is to use `DEBUG_COLORS=no`. This seems to enable the timestamps.

It will be good if winston can be controlled using the DEBUG env var. But we will have to implement a tiny parser here because DEBUG var supports wildcard and "excluding" as well.
